src/NUnitFramework/framework/Api/NUnitTestAssemblyRunner.cs

```diff
@@ -296,7 +296,23 @@ private void StartRun(ITestListener listener)
             if (!System.Diagnostics.Debugger.IsAttached &&
                 Settings.ContainsKey(FrameworkPackageSettings.DebugTests) &&
                 (bool)Settings[FrameworkPackageSettings.DebugTests])
-                System.Diagnostics.Debugger.Launch();
+            {
+                try
+                {
+                    System.Diagnostics.Debugger.Launch();
+                }
+#if !(NETSTANDARD1_3 || NETSTANDARD1_6)
+                catch (SecurityException)
+                {
+                    TopLevelWorkItem.Test.MakeInvalid("System.Security.Permissions.UIPermission is not set to start the debugger.");
+                }
+#endif
+                //System.Diagnostics.Debugger.Launch() not implemented on mono
+                catch (NotImplementedException)
+                {
+                    TopLevelWorkItem.Test.MakeInvalid("Debugger unavailable on this platform.");
```

Comment:
You are making the test assembly invalid but what you really want is to make the result of the assembly invalid. This currently works, because we try to continue running and discover that the assembly is invalid, but it may not continue tl work correctly in the future. I think we actually need to mark the current (top level) __result__ as invalid and return without starting the run.